src/Compilers/CSharp/Test/Emit/PDB/PDBLambdaTests.cs

```diff
@@ -1296,6 +1296,156 @@ void F()
 </symbols>");
         }
 
+        [Fact]
+        public void SwitchExpressionWithLambda1()
+        {
+            string source = WithWindowsLineBreaks(@"
+using System;
+
+class C
+{
+    static string M(object o)
+    {
+        return o switch
+        {
+            int i => new Func<string>(() => $""Number: {i} + {o} == {i + (int)o}"")(),
+            _ => ""Don't know""
+        };
+    }
+}
+");
+
+            // we just want this to compile without crashing/asserting
+            var c = CreateCompilationWithMscorlib40AndSystemCore(source, options: TestOptions.DebugDll);
+            c.VerifyPdb("C.M",
+@"<symbols>
+  <files>
+    <file id=""1"" name="""" language=""C#"" />
+  </files>
+  <methods>
+    <method containingType=""C"" name=""M"" parameterNames=""o"">
+      <customDebugInfo>
+        <using>
+          <namespace usingCount=""1"" />
+        </using>
+        <encLocalSlotMap>
+          <slot kind=""30"" offset=""0"" />
+          <slot kind=""30"" offset=""20"" />
+          <slot kind=""temp"" />
+          <slot kind=""21"" offset=""0"" />
+        </encLocalSlotMap>
+        <encLambdaMap>
+          <methodOrdinal>0</methodOrdinal>
+          <closure offset=""0"" />
+          <closure offset=""20"" />
```

Comment:
><closure offset=""20"" /> [](start = 10, length = 25)

Is offset 20 where the switch expression starts? It would be interesting to have nested switch expressions with closures. Will each switch expression have its own closure?

Reply:
Yes, offset 20 is the start of the switch. This was a good suggestion though, I think maybe it's unearthed another bug. Will comment below.